e2etest/bgptests/bgp.go

```diff
@@ -879,145 +875,122 @@ var _ = ginkgo.Describe("BGP", func() {
+								return fmt.Errorf("%s %s not matching local pref", c.Name, ipWithAdvertisement1)
+							}
+							localPrefix, err = frr.LocalPrefForPrefix(c, ipNoAdvertisement, ipFamily)
+							if err != nil {
+								return err
+							}
+							if localPrefix == *advertisement.LocalPref {
+								return fmt.Errorf("%s %s matching local pref", c.Name, ipNoAdvertisement)
+							}
 
-				for _, c := range FRRContainers {
-					validateFRRPeeredWithNodes(cs, c, family)
+						}
+						return nil
+					}, 1*time.Minute, 1*time.Second).Should(BeNil())
 				}
 
-				ipWithLocalPref, err := config.GetIPFromRangeByIndex(rangeWithPref, 0)
-				framework.ExpectNoError(err)
-				ipWithLocalPref1, err := config.GetIPFromRangeByIndex(rangeWithPref, 1)
-				framework.ExpectNoError(err)
-				ipNoLocalPref, err := config.GetIPFromRangeByIndex(rangeNoPref, 0)
-				framework.ExpectNoError(err)
-				svcLocalPref, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, "svc-local-pref",
-					func(s *corev1.Service) {
-						s.Spec.LoadBalancerIP = ipWithLocalPref
+			},
+			table.Entry("IPV4 - community and localpref",
+				"192.168.10.0/24",
+				"192.168.16.0/24",
+				config.BgpAdvertisement{
+					Communities: []string{
+						CommunityNoAdv,
 					},
-					testservice.TrafficPolicyCluster)
-				defer testservice.Delete(cs, svcLocalPref)
-
-				svcLocalPref1, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, "svc-local-pref1",
-					func(s *corev1.Service) {
-						s.Spec.LoadBalancerIP = ipWithLocalPref1
+					LocalPref: uint32Ptr(50),
+				},
+				ipfamily.IPv4),
+			table.Entry("IPV4 - localpref",
+				"192.168.10.0/24",
+				"192.168.16.0/24",
+				config.BgpAdvertisement{
+					LocalPref: uint32Ptr(50),
+				},
+				ipfamily.IPv4),
+			table.Entry("IPV4 - community",
+				"192.168.10.0/24",
+				"192.168.16.0/24",
+				config.BgpAdvertisement{
+					Communities: []string{
+						CommunityNoAdv,
 					},
-					testservice.TrafficPolicyCluster)
-				defer testservice.Delete(cs, svcLocalPref1)
-
-				svcNoLocalPref, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, "svc-no-local-pref",
-					func(s *corev1.Service) {
-						s.Spec.LoadBalancerIP = ipNoLocalPref
+				},
+				ipfamily.IPv4),
+			table.Entry("IPV6 - community and localpref",
+				"fc00:f853:0ccd:e799::0-fc00:f853:0ccd:e799::18",
+				"fc00:f853:0ccd:e799::19-fc00:f853:0ccd:e799::26",
+				config.BgpAdvertisement{
+					Communities: []string{
+						CommunityNoAdv,
 					},
-					testservice.TrafficPolicyCluster)
-				defer testservice.Delete(cs, svcNoLocalPref)
-
-				allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
-				framework.ExpectNoError(err)
-
-				for _, c := range FRRContainers {
-					validateService(cs, svcLocalPref, allNodes.Items, c)
-					validateService(cs, svcLocalPref1, allNodes.Items, c)
-					validateService(cs, svcNoLocalPref, allNodes.Items, c)
-				}
+					LocalPref: uint32Ptr(50),
+				},
+				ipfamily.IPv6))
```

Comment:
is that intentional that there are no ipv6 localpref only and community only tests?

Reply:
Yes, I wanted to validate the mechanism with ipv4 but not add too much overhead and just make sure the most complex combination works with v6

Reply:
I agree with Sabina here. Are we confident we won't ever regress for IPv6 with community/local preference only?
What is the additional overhead? Test code length? Is it test execution time (how much)?

Also more or less in line, we are not testing the full parameter setting for bgp-advertisements (localPref + communities + aggregation-length). Am I missing something or aggregation length is not being e2e tested?

Reply:
My reasoning was, the rendering part was covered by v4 and the "functional part" was covered by the v4 tests. I wanted to save some test running time, but now that you insist I even believe there are no lanes that run v4 and v6 together (dual runs v4 + explicit DUAL v6 tests). Let me extend that.

Re: aggr length, it's not being tested (I suppose because its a feature that does not depend on the bgp implementation, but better being on the safe side). Let me check if I can add it here.

Reply:
@cgoncalves adding proper aggregation length tests in this context would require a bit of rethinking on how we represent routes in e2e tests. I'd defer this to another pr so we can push this part we know is broken.

Reply:
Added the v6 tests

Reply:
Sure, the aggregation length tests can definitely be addressed outside this PR.
Thanks for the IPv6 tests!